Universal Profiling: on-prem GA additions (#4508)
elastic/observability-docs: docs/en/observability/profiling-get-started.asciidoc

If you're upgrading from a previous version with Universal Profiling enabled, see the <<profiling-upgrade,upgrade guide>>.

[discrete]
[[profiling-configure-data-ingestion-programmatic]]
=== Programmatic configuration

If you prefer to configure data ingestion programmatically, you can use a Kibana API call.

> The API call can be performed either via the "Dev Tools" console in Kibana or any standalone HTTP client (like `curl` or `wget`).
> In both cases, the API call will have tobe executed with the `elastic` user credentials to have the necessary permissions.

Reviewer: ```suggestion
This call can be made either through the "Dev Tools" console in Kibana or with any standalone HTTP client (such as `curl` or `wget`).
In both cases, the API call must be executed using the `elastic` user credentials to ensure the necessary permissions.
```

A successful API call will return a `202 Accepted` response with an empty body.

> To configure data ingestion from the console, go to *Dev Tools* from the navigation menu and execute the following snippet:

Reviewer: ```suggestion
To configure data ingestion from the console, go to *Dev Tools* in the navigation menu and run the following command:

```

[source,console]
----
POST kbn:/internal/profiling/setup/es_resources
{}
----

> To configure data ingestion programmatically using a standalone HTTP client (e.g. `curl`), execute the following command:

Reviewer: ```suggestion
To configure data ingestion programmatically using a standalone HTTP client (e.g., `curl`), run the following command:

```
